### ISSUE TITLE
你好，我想问下我运行训练pnet的时候出现了这个问题#where(): argument 'input' (position 2) must be Tensor, not int

### ISSUE THREAD

**Issue by wbb95** [yeyupiaoling/Pytorch-MTCNN#2]
问题针对这个代码：label = torch.where(label < 0, -100, label)
请问您知道是要把-100转换为tensor吗，只有它是int型变量，此外，您知道如何转换吗

**Comment by yeyupiaoling** [comment]
@wbb95 你的是什么版本？我的不需要特别转换哦。
你试试这样行不行?
```
b = torch.from_numpy(-100)
```

**Comment by wbb95** [comment]
不行哦，新问题：expected np.ndarray (got int)
我也是pytorch,是不是哪个版本跟你一致就好，可以不用改了
由于你设置了label = label.to(device).long()
然后我设置torch.Tensor([-100.]).to(device).long()这样可以
不过又有新问题：在bbox_loss中loss = loss.cuda() * valid_label这一代码
expected device cuda:0 and dtype Float but got device cuda:0 and dtype Long

**Comment by yeyupiaoling** [comment]
Pytorch 1.8.1

**Comment by yeyupiaoling** [comment]
我这个版本使用是没有问题的

**Comment by wbb95** [comment]
我的1.2.0+cu92,就是不知道改成你那个版本后，还会不会有问题

**Comment by yeyupiaoling** [comment]
不会的，

**Comment by wbb95** [comment]
loss = loss.cuda() * valid_label.float()我刚才改了这个也有用
现在已经进行到llandmarkloss,你建议我接着改还是换成你的版本

**Comment by yeyupiaoling** [comment]
是的，不要改代码

**Comment by wbb95** [comment]
我把三个损失函数都改了一遍，只是类型数据的转换
<class 'torch.Tensor'>
[2021-06-18 22:14:57.655575] Train epoch 0, batch 0, total_loss: nan, cls_loss: nan, box_loss: 2.053516, landmarks_loss: 5.919698, accuracy：0.315315
这是目前运行的结果

**Comment by yeyupiaoling** [comment]
分类函数你有没有改错？

**Comment by yeyupiaoling** [comment]
你居然用这么老的版本？

**Comment by wbb95** [comment]
Total params: 6,907
Trainable params: 6,907
Non-trainable params: 0

Input size (MB): 0.00
Forward/backward pass size (MB): 0.02
Params size (MB): 0.03
Estimated Total Size (MB): 0.05
请问这个是pnet参数量是6907个，存储大小是0.03M么，主要我之前看到个有人介绍的pnet网络参数量是27KB，应该对上了，Estimated Total Size (MB): 0.05这个我不清楚是什么

> 应该不会。我只用过 loss = loss.cuda() * valid_label.float()和torch.Tensor([-100.]).to(device).long()

> 怎么，我这个值不大准确么，后面第二个batch准确率0.79

**Comment by yeyupiaoling** [comment]
你看我训练好的模型：https://github.com/yeyupiaoling/Pytorch-MTCNN/tree/master/infer_models

**Comment by yeyupiaoling** [comment]
>Estimated Total Size (MB): 0.05这个我不清楚是什么

这个是Pytorch计算的

**Comment by wbb95** [comment]
嗯嗯，请问你最后训练准确率是多少啊，onet 的识别准确率

**Comment by yeyupiaoling** [comment]
接近100%了。但是这个准确率并不代表什么，最主要的还是框的准。
